Move `current_ability` into it's own module.

eric1234 opened the pull request:
A common need is to render alchemy content within the another context (the app or another engine). An example of this would be the alchemy-solidus gem. We do this by included `Alchemy::ControllerActions` into the controllers for the context we want to use the Alchemy content in.

The problem is when this is done IF that gem is also using cancancan (like Solidus does) then the `current_ability` method from Alchemy can end up overwritting the `current_ability` of the controller doing the including.

By moving `current_ability` to it's own module it allows the including controller to decide if that method is desired or not.

Motivation for this commit is driven from AlchemyCMS/alchemy-solidus#6 to simplify that PR.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://eric1234-separate-auth-module.alchemycms-alchemy-cms-42e8f4fc945384e0f6b5.ttrcloud.com/admin' target='_blank'>Open stage</a> or <a href='http://eric1234-separate-auth-module.alchemycms-alchemy-cms-42e8f4fc945384e0f6b5.ttrcloud.com/admin/teatro.log' target='_blank'>view logs</a>.

tvdeyen commented:
👍 thank you!

tvdeyen commented:
@eric1234 could you please add a CHANGELOG entry and squash the commits into one? And maybe we should document that if used outside of Alchemy it's ensured that a `current_alchemy_user` method exists otherwise we get "undefined method" errors. Thanks

eric1234 commented:
I have added to the changelog and squashed the commits.

As to the note about `undefined method` I'm not sure that is necessary. Anybody using `Alchemy::AbilityHelper` would also be using `Alchemy::ControllerActions` (which defines `current_alchemy_user`). The goal is to be able to use `Alchemy::ControllerActions` without `current_ability`. The other way around would just not be something that should be supported. If they do for some reason then they have gone in the deep end on their own.

tvdeyen commented:
👍 I'm fine with this @mamhoff want to add your 2 cents?

mamhoff commented:
No two cents here - this is just good :) 👍

tvdeyen commented:
Thanks!
